### PR TITLE
Refs #23919 -- Removed six.<various>_types usage

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: django/db/models/fields/__init__.py [hunk 1]
```diff
@@ -244,7 +244,7 @@ def _check_field_name(self):
 
     def _check_choices(self):
         if self.choices:
-            if (isinstance(self.choices, six.string_types) or
+            if (isinstance(self.choices, str) or
```
Comment (timgraham) [inline django/db/models/fields/__init__.py]: single line looks okay here

File: django/urls/resolvers.py [hunk 2]
```diff
@@ -104,7 +104,7 @@ def _compile(self, regex):
         except re.error as e:
             raise ImproperlyConfigured(
                 '"%s" is not a valid regular expression: %s' %
-                (regex, six.text_type(e))
+                (regex, str(e))
```
Comment (timgraham) [inline django/urls/resolvers.py]: Does `str()` do something differently from `%s`?

File: django/utils/encoding.py [hunk 3]
```diff
@@ -36,8 +36,8 @@ def smart_text(s, encoding='utf-8', strings_only=False, errors='strict'):
     return force_text(s, encoding, strings_only, errors)
 
 
-_PROTECTED_TYPES = six.integer_types + (
-    type(None), float, Decimal, datetime.datetime, datetime.date, datetime.time
+_PROTECTED_TYPES = (
+    type(None), int, float, Decimal, datetime.datetime, datetime.date, datetime.time
```
Comment (timgraham) [inline django/utils/encoding.py]: could add the trailing comma there

File: django/contrib/auth/models.py [hunk 4]
```diff
@@ -75,9 +75,9 @@ class Meta:
 
     def __str__(self):
         return "%s | %s | %s" % (
-            six.text_type(self.content_type.app_label),
-            six.text_type(self.content_type),
-            six.text_type(self.name))
+            self.content_type.app_label,
+            str(self.content_type),
+            self.name)
```
Comment (charettes) [inline django/contrib/auth/models.py]: Might want to move the `)` to the next line and add a trailing comma when you are around.

File: tests/admin_views/tests.py [hunk 5]
```diff
@@ -3759,7 +3759,7 @@ def test_edit_model_modeladmin_defer_qs(self):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(ShortMessage.objects.count(), 1)
         # Message should contain non-ugly model verbose name. The ugly(!)
-        # instance representation is set by six.text_type()
+        # instance representation is set by str().
```
Comment (timgraham) [inline tests/admin_views/tests.py]: I think this is actually referring to `__str__()`

File: tests/httpwrappers/tests.py [hunk 6]
```diff
@@ -617,7 +617,7 @@ def test_streaming_response(self):
         # message like a regular response does. it only gives us the headers.
         r = StreamingHttpResponse(iter(['hello', 'world']))
         self.assertEqual(
```
Comment (timgraham) [inline tests/httpwrappers/tests.py]: single line here

File: django/contrib/auth/models.py [hunk 7]
```diff
@@ -75,9 +75,9 @@ class Meta:
 
     def __str__(self):
         return "%s | %s | %s" % (
-            six.text_type(self.content_type.app_label),
-            six.text_type(self.content_type),
-            six.text_type(self.name))
+            self.content_type.app_label,
+            str(self.content_type),
```
Comment (claudep) [inline django/contrib/auth/models.py]: yes, str() is needed to force content_type *object* to a string, while other values are (AFAIK) strings already.

Comment (charettes) [inline django/contrib/auth/models.py]: But isn't the `'%s'` format specifier going to call `__str__` anyway?

Comment (claudep) [inline django/contrib/auth/models.py]: Mmh, yes, that's TRUE!

File: tests/admin_views/tests.py [hunk 8]
```diff
@@ -3806,7 +3806,7 @@ def test_edit_model_modeladmin_only_qs(self):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(Paper.objects.count(), 1)
         # Message should contain non-ugly model verbose name. The ugly(!)
-        # instance representation is set by six.text_type()
+        # instance representation is set by str().
```
Comment (timgraham) [inline tests/admin_views/tests.py]: (as above)

**Pull request closed without merging**